addon/-private/system/relationships/state/create.js

```diff
@@ -57,23 +57,19 @@ export default class Relationships {
     if (!relationship) {
       let relationshipsByName = get(internalModel.type, 'relationshipsByName');
       let rel = relationshipsByName.get(key);
-      let relationshipPayloadEntry = internalModel.store._relationshipsPayloads.get(internalModel.modelName, internalModel.id, key);
+      let relationshipPayload = internalModel.store._relationshipsPayloads.get(internalModel.modelName, internalModel.id, key);
 
-      if (rel) {
-        relationship = relationships[key] = createRelationshipFor(internalModel, rel, internalModel.store);
-      }
+      if (!rel) { return undefined; }
+
+      relationship = relationships[key] = createRelationshipFor(internalModel, rel, internalModel.store);
 
-      if (relationshipPayloadEntry !== undefined) {
-        let { payload, inverse } = relationshipPayloadEntry;
-        if (!inverse) {
-          relationship.push(payload);
-        } else {
-          relationship.pushFromInverse(payload);
-        }
+      if (relationshipPayload) {
+        this.internalModel.store._backburner.join(function() {
```

Comment:
This is the thing that batches canonical flushes.  In master this batching happens [within _push](https://github.com/emberjs/data/blob/03440c14e14ab7923ae2ed7db8fbbec6d3fa89e6/addon/-private/system/store.js#L2315-L2342)